gobblin-runtime/src/main/java/gobblin/runtime/AbstractTaskStateTracker.java

```diff
@@ -105,6 +108,7 @@ public TaskMetricsUpdater(Task task) {
 
     @Override
     public void run() {
+      MDC.put(ConfigurationKeys.TASK_KEY_KEY, task.getTaskKey());
```

Comment:
Correct.  When a `Callable` or `Runnable` are run as part of `MDCPropagatingScheduledExecutorService` or `MDCPropagatingExecutorService`, the class is wrapped with code that automatically clone and disposes if the context.  Anything you set into the MDC in this case should automatically go away.